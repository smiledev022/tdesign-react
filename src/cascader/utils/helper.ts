import { TreeNode } from '../interface';

/**
 * 获取node节点的全路径
 * @param node
 * @returns
 */
export function getFullPathLabel(node: TreeNode) {
  return node
    .getPath()
    .map((node: TreeNode) => node.label)
    .join('/');
}
