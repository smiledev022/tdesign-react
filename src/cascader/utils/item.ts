import { CascaderContextType, TreeNode } from '../interface';

/**
 * 是否显示省略计算方法
 * @param node
 * @param size
 * @returns
 */
export function getLabelIsEllipsis(node: TreeNode, size: CascaderContextType['size']) {
  const sizeMap = {
    small: 11,
    medium: 9,
    large: 8,
  };
  return sizeMap[size] < node.label.length;
}

export function getCascaderItemClass(
  prefix: string,
  node: TreeNode,
  CLASSNAMES: any,
  cascaderContext: CascaderContextType,
) {
  const { checkStrictly, multiple, size } = cascaderContext;
  const expandedActive = !checkStrictly && node.expanded && (multiple ? !node.isLeaf() : true);

  return [
    `${prefix}-cascader-item`,
    {
      [CLASSNAMES.STATUS.selected]: node.checked,
      [CLASSNAMES.STATUS.expanded]: expandedActive,
      [CLASSNAMES.STATUS.disabled]: node.disabled,
      [CLASSNAMES.STATUS.active]: node.actived || (multiple && node.expanded),
      [CLASSNAMES.SIZE[size]]: size,
      [`${prefix}-cascader-item-have-icon`]: node.children,
    },
  ];
}

export function getCascaderItemIconClass(
  prefix: string,
  node: TreeNode,
  CLASSNAMES: any,
  cascaderContext: CascaderContextType,
) {
  const { checkStrictly } = cascaderContext;

  return [
    `${prefix}-cascader-icon`,
    {
      [CLASSNAMES.STATUS.expanded]: !checkStrictly && node.expanded,
    },
  ];
}
