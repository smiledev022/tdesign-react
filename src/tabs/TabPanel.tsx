import React from 'react';
import { TabPanelProps } from './TabProps';

const TabPanel: React.FC<TabPanelProps> = (props) => {
  const { active, forceRender } = props;
  if (forceRender) {
    return (
      <div
        className={'t-tab-panel'}
        style={{
          display: active ? 'block' : 'none',
        }}
      >
        {props.children}
      </div>
    );
  }
  return active && <div className={'t-tab-panel'}>{props.children}</div>;
};

TabPanel.displayName = 'TabPanel';

export default TabPanel;
