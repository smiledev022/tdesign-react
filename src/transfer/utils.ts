import isEmpty from 'lodash/isEmpty';
import isFunction from 'lodash/isFunction';
import isString from 'lodash/isString';
import React from 'react';
import { TNode } from '../common';

export const getDefaultValue = <T>(value: T | T[], defaultValue?: T): T[] => {
  if (isEmpty(value)) {
    return Array.isArray(defaultValue) ? defaultValue : [defaultValue, defaultValue];
  }

  if (Array.isArray(value)) return value;

  return [defaultValue, defaultValue];
};

export const getJSX = (value: string | TNode) => {
  if (isString(value)) return value;

  if (isFunction(value)) return value();

  if (React.isValidElement(value)) return value;

  return value;
};
