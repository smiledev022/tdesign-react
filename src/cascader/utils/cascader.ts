import { getFullPathLabel } from './helper';
import { TreeNode, TreeNodeValue, TreeOptionData, CascaderContextType } from '../interface';

/**
 * treeValue计算方法
 * @param model
 * @returns
 */
export const getTreeValue = (model: CascaderContextType['model']) => {
  let treeValue: TreeNodeValue[] = [];
  if (Array.isArray(model)) {
    if (model.length > 0 && typeof model[0] === 'object') {
      treeValue = (model as TreeOptionData[]).map((val) => val.value);
    } else if (model.length) {
      treeValue = model as TreeNodeValue[];
    }
  } else if (model) {
    if (typeof model === 'object') {
      treeValue = [(model as TreeOptionData).value];
    } else {
      treeValue = [model];
    }
  }
  return treeValue;
};

/**
 * input和treeStore变化的副作用
 * @param inputVal
 * @param treeStore
 * @param setTreeNodes
 * @returns
 */
export const treeNodesEffect = (
  inputVal: CascaderContextType['inputVal'],
  treeStore: CascaderContextType['treeStore'],
  setTreeNodes: CascaderContextType['setTreeNodes'],
) => {
  if (!treeStore) return;
  let nodes = [];
  if (inputVal) {
    nodes = treeStore.nodes.filter((node: TreeNode) => {
      const fullPathLabel = getFullPathLabel(node);
      return fullPathLabel.indexOf(inputVal) > -1 && node.isLeaf();
    });
  } else {
    nodes = treeStore.getNodes().filter((node: TreeNode) => node.visible);
  }
  setTreeNodes(nodes);
};

/**
 * 初始化展开阶段与展开状态副作用
 * @param treeStore
 * @param treeValue
 * @param expend
 */
export const treeStoreExpendEffect = (
  treeStore: CascaderContextType['treeStore'],
  treeValue: TreeNodeValue[],
  expend: TreeNodeValue[],
) => {
  // init expanded, 无expend状态时设置
  if (Array.isArray(treeValue) && expend.length === 0) {
    const expandedMap = new Map();
    const [val] = treeValue;
    if (val) {
      expandedMap.set(val, true);
      const node = treeStore.getNode(val);
      if (!node) return;
      node.getParents().forEach((tn: TreeNode) => {
        expandedMap.set(tn.value, true);
      });
      const expandedArr = Array.from(expandedMap.keys());
      treeStore.replaceExpanded(expandedArr);
    }
  }
  // 本地维护 expend，更加可控，不需要依赖于 tree 的状态
  if (treeStore.getExpanded() && expend.length) {
    treeStore.replaceExpanded(expend);
  }
  treeStore.refreshNodes();
};
