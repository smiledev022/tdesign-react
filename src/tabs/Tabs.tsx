import React, { useState } from 'react';
import classNames from 'classnames';
import { TabValue, TdTabsProps } from '../_type/components/tabs';
import TabNav from './TabNav';
import { useTabClass } from './useTabClass';
import TabPanel from './TabPanel';

export interface TabsProps extends TdTabsProps {}

const Tabs: React.FC<TabsProps> = (props) => {
  const { children, defaultValue, placement, onRemove } = props;

  // 样式工具引入
  const { tdTabPanelClassPrefix, tdTabsClassPrefix, tdTabsClassGenerator, tdClassGenerator } = useTabClass();

  const [value, setValue] = useState<TabValue>(defaultValue);

  const itemList = React.Children.map(children, (child: any) => {
    if (child && child.type === TabPanel) {
      return child.props;
    }
    return null;
  });

  const renderTabNav = () => (
    <TabNav {...props} activeValue={value} onRemove={onRemove} itemList={itemList} tabClick={setValue} />
  );

  return (
    <div className={classNames(tdTabsClassPrefix)}>
      {placement !== 'bottom' ? renderTabNav() : null}
      <div className={classNames(tdTabsClassGenerator('content'), tdClassGenerator(`is-${placement}`))}>
        <div className={classNames(tdTabPanelClassPrefix)}>
          {React.Children.map(children, (child: any) => {
            if (child && child.type === TabPanel && child.props.value === value) {
              return child;
            }
            return null;
          })}
        </div>
      </div>
      {placement === 'bottom' ? renderTabNav() : null}
    </div>
  );
};

Tabs.displayName = 'Tabs';

export default Tabs;
