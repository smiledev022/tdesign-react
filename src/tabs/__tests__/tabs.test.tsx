import React from 'react';
import { render, testExamples, waitFor } from '@test/utils';
import TabPanel from '../TabPanel';
import Tabs from '../Tabs';

testExamples(__dirname);

describe('Tabs 组件测试', () => {
  test('render Tab bar', async () => {
    const testId = 'tab bar test id';
    const { getByTestId } = render(
      <div data-testid={testId}>
        <Tabs placement={'top'} data-testid={testId} size={'medium'}>
          <TabPanel value={'a'} label={'a'}>
            <div>a</div>
          </TabPanel>
          <TabPanel value={'b'} label={'b'}>
            <div>b</div>
          </TabPanel>
        </Tabs>
      </div>,
    );

    const tabInstance = await waitFor(() => getByTestId(testId));

    expect(() => tabInstance.querySelector('.t-tabs__nav')).not.toBe(null);
  });

  test('render card theme', async () => {
    const testId = 'tab card theme test id';
    const { getByTestId } = render(
      <div data-testid={testId}>
        <Tabs placement={'top'} size={'medium'}>
          <TabPanel value={'a'} label={'a'}>
            <div>a</div>
          </TabPanel>
          <TabPanel value={'b'} label={'b'}>
            <div>b</div>
          </TabPanel>
        </Tabs>
      </div>,
    );

    const tabInstance = await waitFor(() => getByTestId(testId));

    expect(() => tabInstance.querySelector('.t-tabs__nav--card')).not.toBe(null);
  });

  test('different position', async () => {
    const testId = 'tab position test id';
    const { getByTestId } = render(
      <div data-testid={testId}>
        <Tabs placement={'top'} size={'medium'}>
          <TabPanel value={'a'} label={'a'}>
            <div>a</div>
          </TabPanel>
          <TabPanel value={'b'} label={'b'}>
            <div>b</div>
          </TabPanel>
        </Tabs>
      </div>,
    );

    const tabInstance = await waitFor(() => getByTestId(testId));

    expect(() => tabInstance.querySelector('.t-is-top')).not.toBe(null);
  });
});
