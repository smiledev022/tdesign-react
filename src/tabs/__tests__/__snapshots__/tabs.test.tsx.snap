// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddTabs.jsx 1`] = `
<DocumentFragment>
  <div
    style="max-width: 400px;"
  >
    <div
      class="t-tabs"
    >
      <div
        class="t-tabs__header t-is-top"
      >
        <div
          class="t-tabs__nav"
        >
          <span
            class="t-tabs__add-btn t-size-m"
          >
            +
          </span>
          <div
            class="t-tabs__nav-container t-is-top t-is-addable"
          >
            <div
              class="t-tabs__nav-scroll"
            >
              <div
                class="t-tabs__nav-wrap"
                style="transform: translateX(0px);"
              >
                <div
                  class="t-tabs__bar t-is-top"
                  style="transform: translateX(0px);"
                />
                <div
                  class="t-tabs__nav-item t-tabs__nav--card t-is-active t-size-m"
                >
                  1
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="t-tabs__content"
      >
        <div
          class="t-tab-panel"
        >
          <div
            style="margin: 20px;"
          >
            1
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`BasicTabs.jsx 1`] = `
<DocumentFragment>
  <div
    style="max-width: 400px;"
  >
    <div
      class="t-tabs"
    >
      <div
        class="t-tabs__header t-is-top"
      >
        <div
          class="t-tabs__nav"
        >
          <div
            class="t-tabs__nav-container t-is-top"
          >
            <div
              class="t-tabs__nav-scroll"
            >
              <div
                class="t-tabs__nav-wrap"
                style="transform: translateX(0px);"
              >
                <div
                  class="t-tabs__bar t-is-top"
                  style="transform: translateX(0px);"
                />
                <div
                  class="t-tabs__nav-item t-is-active t-size-m"
                >
                  a
                </div>
                <div
                  class="t-tabs__nav-item t-size-m"
                >
                  b
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="t-tabs__content"
      >
        <div
          class="t-tab-panel"
        >
          <div
            style="margin: 20px;"
          >
            a
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`CloseableTabs.jsx 1`] = `
<DocumentFragment>
  <div
    style="max-width: 400px;"
  >
    <div
      class="t-tabs"
    >
      <div
        class="t-tabs__header t-is-top"
      >
        <div
          class="t-tabs__nav"
        >
          <span
            class="t-tabs__add-btn t-size-m"
          >
            +
          </span>
          <div
            class="t-tabs__nav-container t-is-top t-is-addable"
          >
            <div
              class="t-tabs__nav-scroll"
            >
              <div
                class="t-tabs__nav-wrap"
                style="transform: translateX(0px);"
              >
                <div
                  class="t-tabs__bar t-is-top"
                  style="transform: translateX(0px);"
                />
                <div
                  class="t-tabs__nav-item t-tabs__nav--card t-is-active t-size-m"
                >
                  1
                  <svg
                    class="t-icon t-icon-close"
                    version="1.1"
                    viewBox="0 0 1024 1024"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M760.9780096 217.6L806.2328448 262.85483392 557.312 511.744 806.2328448 760.6580096 760.9780096 805.9128448 512.064 556.992 263.17483392 805.9128448 217.92 760.6580096 466.816 511.744 217.92 262.85483392 263.17483392 217.6 512.064 466.496 760.9780096 217.6Z"
                      fill-opacity=".9"
                      fill-rule="evenodd"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="t-tabs__content"
      >
        <div
          class="t-tab-panel"
        >
          <div
            style="margin: 20px;"
          >
            1
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`PositionTabs.jsx 1`] = `
<DocumentFragment>
  <div>
    <button
      class="t-button t-button--theme-default t-button--variant-base"
    >
      top
    </button>
    <button
      class="t-button t-button--theme-default t-button--variant-base"
    >
      bottom
    </button>
    <button
      class="t-button t-button--theme-default t-button--variant-base"
    >
      left
    </button>
    <button
      class="t-button t-button--theme-default t-button--variant-base"
    >
      right
    </button>
    <div
      class="t-tabs"
    >
      <div
        class="t-tabs__header t-is-top"
      >
        <div
          class="t-tabs__nav"
        >
          <div
            class="t-tabs__nav-container t-is-top t-is-addable"
          >
            <div
              class="t-tabs__nav-scroll"
            >
              <div
                class="t-tabs__nav-wrap"
                style="transform: translateX(0px);"
              >
                <div
                  class="t-tabs__bar t-is-top"
                  style="transform: translateX(0px);"
                />
                <div
                  class="t-tabs__nav-item t-is-active t-size-m"
                >
                  1
                </div>
                <div
                  class="t-tabs__nav-item t-size-m"
                >
                  <div>
                    2
                  </div>
                </div>
                <div
                  class="t-tabs__nav-item t-size-m"
                >
                  3
                </div>
                <div
                  class="t-tabs__nav-item t-size-m"
                >
                  4
                </div>
                <div
                  class="t-tabs__nav-item t-size-m"
                >
                  <div>
                    5
                  </div>
                </div>
                <div
                  class="t-tabs__nav-item t-size-m"
                >
                  <div>
                    6
                  </div>
                </div>
                <div
                  class="t-tabs__nav-item t-size-m"
                >
                  <div>
                    7
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="t-tabs__content"
      >
        <div
          class="t-tab-panel"
          style="display: block;"
        >
          <div
            style="margin: 20px;"
          >
            这是一个Tabs
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`SizeTabs.jsx 1`] = `
<DocumentFragment>
  <div>
    <button
      class="t-button t-button--theme-default t-button--variant-base"
    >
      small
    </button>
    <button
      class="t-button t-button--theme-default t-button--variant-base"
    >
      middle
    </button>
    <button
      class="t-button t-button--theme-default t-button--variant-base"
    >
      large
    </button>
    <div
      class="t-tabs"
    >
      <div
        class="t-tabs__header t-is-top"
      >
        <div
          class="t-tabs__nav"
        >
          <div
            class="t-tabs__nav-container t-is-top t-is-addable"
          >
            <div
              class="t-tabs__nav-scroll"
            >
              <div
                class="t-tabs__nav-wrap"
                style="transform: translateX(0px);"
              >
                <div
                  class="t-tabs__bar t-is-top"
                  style="transform: translateX(0px);"
                />
                <div
                  class="t-tabs__nav-item t-is-active t-size-m"
                >
                  1
                </div>
                <div
                  class="t-tabs__nav-item t-size-m"
                >
                  <div>
                    2
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="t-tabs__content"
      >
        <div
          class="t-tab-panel"
          style="display: block;"
        >
          <div
            style="margin: 20px;"
          >
            这是一个Tabs
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ThemeTabs.jsx 1`] = `
<DocumentFragment>
  <div>
    <button
      class="t-button t-button--theme-default t-button--variant-base"
    >
      default
    </button>
    <button
      class="t-button t-button--theme-default t-button--variant-base"
    >
      card
    </button>
    <div
      class="t-tabs"
    >
      <div
        class="t-tabs__header t-is-top"
      >
        <div
          class="t-tabs__nav"
        >
          <div
            class="t-tabs__nav-container t-is-top"
          >
            <div
              class="t-tabs__nav-scroll"
            >
              <div
                class="t-tabs__nav-wrap"
                style="transform: translateX(0px);"
              >
                <div
                  class="t-tabs__bar t-is-top"
                  style="transform: translateX(0px);"
                />
                <div
                  class="t-tabs__nav-item t-is-active t-size-m"
                >
                  1
                </div>
                <div
                  class="t-tabs__nav-item t-size-m"
                >
                  <div>
                    2
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="t-tabs__content"
      >
        <div
          class="t-tab-panel"
        >
          <div
            style="margin: 20px;"
          >
            这是一个Tabs
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`base.jsx 1`] = `
<DocumentFragment>
  <div
    class="tdegsin-demo-tabs"
  >
    <div
      class="t-tabs"
    >
      <div
        class="t-tabs__header t-is-top"
      >
        <div
          class="t-tabs__nav"
        >
          <div
            class="t-tabs__nav-container t-is-top"
          >
            <div
              class="t-tabs__nav-scroll"
            >
              <div
                class="t-tabs__nav-wrap"
                style="transform: translateX(0px);"
              >
                <div
                  class="t-tabs__bar t-is-top"
                  style="transform: translateX(0px);"
                />
                <div
                  class="t-tabs__nav-item t-is-active t-size-m"
                >
                  选项卡1
                </div>
                <div
                  class="t-tabs__nav-item t-size-m"
                >
                  选项卡2
                </div>
                <div
                  class="t-tabs__nav-item t-size-m"
                >
                  选项卡3
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="t-tabs__content"
      >
        <div
          class="t-tab-panel"
        >
          <div
            class="tabs-content"
          >
            选项卡1
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`combination.jsx 1`] = `
<DocumentFragment>
  <div
    class="tdegsin-demo-tabs"
  >
    <div
      class="t-tabs"
    >
      <div
        class="t-tabs__header t-is-top"
      >
        <div
          class="t-tabs__nav"
        >
          <span
            class="t-tabs__add-btn t-size-m"
          >
            +
          </span>
          <div
            class="t-tabs__nav-container t-is-top t-is-addable"
          >
            <div
              class="t-tabs__nav-scroll"
            >
              <div
                class="t-tabs__nav-wrap"
                style="transform: translateX(0px);"
              >
                <div
                  class="t-tabs__bar t-is-top"
                  style="transform: translateX(0px);"
                />
                <div
                  class="t-tabs__nav-item t-tabs__nav--card t-is-active t-size-m"
                >
                  选项卡1
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="t-tabs__content"
      >
        <div
          class="t-tab-panel"
        >
          <div
            class="tabs-content"
          >
            选项卡1
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`custom.jsx 1`] = `
<DocumentFragment>
  <div
    class="tdegsin-demo-tabs"
  >
    <div
      class="t-tabs"
    >
      <div
        class="t-tabs__header t-is-top"
      >
        <div
          class="t-tabs__nav"
        >
          <span
            class="t-tabs__add-btn t-size-m"
          >
            +
          </span>
          <div
            class="t-tabs__nav-container t-is-top t-is-addable"
          >
            <div
              class="t-tabs__nav-scroll"
            >
              <div
                class="t-tabs__nav-wrap"
                style="transform: translateX(0px);"
              >
                <div
                  class="t-tabs__bar t-is-top"
                  style="transform: translateX(0px);"
                />
                <div
                  class="t-tabs__nav-item t-tabs__nav--card t-is-active t-size-m"
                >
                  选项卡1
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="t-tabs__content"
      >
        <div
          class="t-tab-panel"
        >
          <div
            class="tabs-content"
          >
            选项卡1
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`icon.jsx 1`] = `
<DocumentFragment>
  <div
    class="tdegsin-demo-tabs"
  >
    <button
      class="t-button t-button--theme-default t-button--variant-outline"
    >
      切换到卡片型
    </button>
    <div
      class="t-tabs"
    >
      <div
        class="t-tabs__header t-is-top"
      >
        <div
          class="t-tabs__nav"
        >
          <div
            class="t-tabs__nav-container t-is-top"
          >
            <div
              class="t-tabs__nav-scroll"
            >
              <div
                class="t-tabs__nav-wrap"
                style="transform: translateX(0px);"
              >
                <div
                  class="t-tabs__bar t-is-top"
                  style="transform: translateX(0px);"
                />
                <div
                  class="t-tabs__nav-item t-is-active t-size-m"
                >
                  <div>
                    <svg
                      class="t-icon t-icon-discount"
                      version="1.1"
                      viewBox="0 0 1024 1024"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M493.04229568 116.1597152C508.86786752 116.30420928 524.0067104 121.97761344 535.39539904 133.36630208L942.6889024 540.659808C967.6824576 565.65336256 965.1497984 608.70858112 937.0320512 636.82633024L631.56192128 942.2964608C603.44417216 970.414208 560.3889536 972.9468672 535.39539904 947.953312L128.10189312 540.659808C116.179456 528.73737088 110.52056832 512.70502848 110.91453056 496.14468224L110.89530624 496.1568576 110.89530624 180.1597152C110.89530624 147.33822144 135.6018784 120.28725696 167.43155456 116.59028992L174.89530624 116.1597152 493.04229568 116.1597152ZM174.89530624 180.0957152L174.89530624 496.1568576 174.06330624 496.1277152 580.65023296 902.6984768 897.4340736 585.91464192 491.69530624 180.1597152 174.89530624 180.0957152ZM396.1453088 256.40588416C466.8377568 256.40588416 524.1453088 313.71343616 524.1453088 384.40588416 524.1453088 455.09833216 466.8377568 512.40588416 396.1453088 512.40588416 325.4528608 512.40588416 268.1453088 455.09833216 268.1453088 384.40588416 268.1453088 313.71343616 325.4528608 256.40588416 396.1453088 256.40588416ZM396.1453088 320.40588416C360.7990848 320.40588416 332.1453088 349.05966016 332.1453088 384.40588416 332.1453088 419.75210816 360.7990848 448.40588416 396.1453088 448.40588416 431.4915328 448.40588416 460.1453088 419.75210816 460.1453088 384.40588416 460.1453088 349.05966016 431.4915328 320.40588416 396.1453088 320.40588416Z"
                        fill-opacity=".9"
                        fill-rule="evenodd"
                      />
                    </svg>
                    选项卡
                  </div>
                </div>
                <div
                  class="t-tabs__nav-item t-size-m"
                >
                  <div>
                    <svg
                      class="t-icon t-icon-discount"
                      version="1.1"
                      viewBox="0 0 1024 1024"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M493.04229568 116.1597152C508.86786752 116.30420928 524.0067104 121.97761344 535.39539904 133.36630208L942.6889024 540.659808C967.6824576 565.65336256 965.1497984 608.70858112 937.0320512 636.82633024L631.56192128 942.2964608C603.44417216 970.414208 560.3889536 972.9468672 535.39539904 947.953312L128.10189312 540.659808C116.179456 528.73737088 110.52056832 512.70502848 110.91453056 496.14468224L110.89530624 496.1568576 110.89530624 180.1597152C110.89530624 147.33822144 135.6018784 120.28725696 167.43155456 116.59028992L174.89530624 116.1597152 493.04229568 116.1597152ZM174.89530624 180.0957152L174.89530624 496.1568576 174.06330624 496.1277152 580.65023296 902.6984768 897.4340736 585.91464192 491.69530624 180.1597152 174.89530624 180.0957152ZM396.1453088 256.40588416C466.8377568 256.40588416 524.1453088 313.71343616 524.1453088 384.40588416 524.1453088 455.09833216 466.8377568 512.40588416 396.1453088 512.40588416 325.4528608 512.40588416 268.1453088 455.09833216 268.1453088 384.40588416 268.1453088 313.71343616 325.4528608 256.40588416 396.1453088 256.40588416ZM396.1453088 320.40588416C360.7990848 320.40588416 332.1453088 349.05966016 332.1453088 384.40588416 332.1453088 419.75210816 360.7990848 448.40588416 396.1453088 448.40588416 431.4915328 448.40588416 460.1453088 419.75210816 460.1453088 384.40588416 460.1453088 349.05966016 431.4915328 320.40588416 396.1453088 320.40588416Z"
                        fill-opacity=".9"
                        fill-rule="evenodd"
                      />
                    </svg>
                    选项卡
                  </div>
                </div>
                <div
                  class="t-tabs__nav-item t-size-m"
                >
                  <div>
                    <svg
                      class="t-icon t-icon-discount"
                      version="1.1"
                      viewBox="0 0 1024 1024"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M493.04229568 116.1597152C508.86786752 116.30420928 524.0067104 121.97761344 535.39539904 133.36630208L942.6889024 540.659808C967.6824576 565.65336256 965.1497984 608.70858112 937.0320512 636.82633024L631.56192128 942.2964608C603.44417216 970.414208 560.3889536 972.9468672 535.39539904 947.953312L128.10189312 540.659808C116.179456 528.73737088 110.52056832 512.70502848 110.91453056 496.14468224L110.89530624 496.1568576 110.89530624 180.1597152C110.89530624 147.33822144 135.6018784 120.28725696 167.43155456 116.59028992L174.89530624 116.1597152 493.04229568 116.1597152ZM174.89530624 180.0957152L174.89530624 496.1568576 174.06330624 496.1277152 580.65023296 902.6984768 897.4340736 585.91464192 491.69530624 180.1597152 174.89530624 180.0957152ZM396.1453088 256.40588416C466.8377568 256.40588416 524.1453088 313.71343616 524.1453088 384.40588416 524.1453088 455.09833216 466.8377568 512.40588416 396.1453088 512.40588416 325.4528608 512.40588416 268.1453088 455.09833216 268.1453088 384.40588416 268.1453088 313.71343616 325.4528608 256.40588416 396.1453088 256.40588416ZM396.1453088 320.40588416C360.7990848 320.40588416 332.1453088 349.05966016 332.1453088 384.40588416 332.1453088 419.75210816 360.7990848 448.40588416 396.1453088 448.40588416 431.4915328 448.40588416 460.1453088 419.75210816 460.1453088 384.40588416 460.1453088 349.05966016 431.4915328 320.40588416 396.1453088 320.40588416Z"
                        fill-opacity=".9"
                        fill-rule="evenodd"
                      />
                    </svg>
                    选项卡
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="t-tabs__content"
      >
        <div
          class="t-tab-panel"
        >
          <div
            class="tabs-content"
          >
            选项卡1
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`operation.jsx 1`] = `
<DocumentFragment>
  <div
    style="max-width: 400px;"
  >
    <div
      class="t-tabs"
    >
      <div
        class="t-tabs__header t-is-top"
      >
        <div
          class="t-tabs__nav"
        >
          <span
            class="t-tabs__add-btn t-size-m"
          >
            +
          </span>
          <div
            class="t-tabs__nav-container t-is-top t-is-addable"
          >
            <div
              class="t-tabs__nav-scroll"
            >
              <div
                class="t-tabs__nav-wrap"
                style="transform: translateX(0px);"
              >
                <div
                  class="t-tabs__bar t-is-top"
                  style="transform: translateX(0px);"
                />
                <div
                  class="t-tabs__nav-item t-tabs__nav--card t-is-active t-size-m"
                >
                  选项卡1
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="t-tabs__content"
      >
        <div
          class="t-tab-panel"
        >
          <div
            style="margin: 20px;"
          >
            选项卡1
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`position.jsx 1`] = `
<DocumentFragment>
  <div
    class="tdegsin-demo-tabs"
  >
    <button
      class="t-button t-button--theme-default t-button--variant-base"
    >
      top
    </button>
    <button
      class="t-button t-button--theme-default t-button--variant-base"
    >
      bottom
    </button>
    <button
      class="t-button t-button--theme-default t-button--variant-base"
    >
      left
    </button>
    <button
      class="t-button t-button--theme-default t-button--variant-base"
    >
      right
    </button>
    <div
      class="t-tabs"
    >
      <div
        class="t-tabs__header t-is-top"
      >
        <div
          class="t-tabs__nav"
        >
          <div
            class="t-tabs__nav-container t-is-top t-is-addable"
          >
            <div
              class="t-tabs__nav-scroll"
            >
              <div
                class="t-tabs__nav-wrap"
                style="transform: translateX(0px);"
              >
                <div
                  class="t-tabs__bar t-is-top"
                  style="transform: translateX(0px);"
                />
                <div
                  class="t-tabs__nav-item t-is-active t-size-m"
                >
                  选项卡1
                </div>
                <div
                  class="t-tabs__nav-item t-size-m"
                >
                  <div>
                    选项卡2
                  </div>
                </div>
                <div
                  class="t-tabs__nav-item t-size-m"
                >
                  选项卡3
                </div>
                <div
                  class="t-tabs__nav-item t-size-m"
                >
                  选项卡4
                </div>
                <div
                  class="t-tabs__nav-item t-size-m"
                >
                  <div>
                    选项卡5
                  </div>
                </div>
                <div
                  class="t-tabs__nav-item t-size-m"
                >
                  <div>
                    选项卡6
                  </div>
                </div>
                <div
                  class="t-tabs__nav-item t-size-m"
                >
                  <div>
                    选项卡7
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="t-tabs__content"
      >
        <div
          class="t-tab-panel"
          style="display: block;"
        >
          <div
            class="tabs-content"
          >
            这是一个Tabs
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
